input/postgres/collection_helper.go

```diff
@@ -31,11 +35,37 @@ func NewCollectionHelper(ctx context.Context, logger *util.Logger, server *state
 	}
 	server.SelfTest.MarkCollectionAspect(state.CollectionAspectPgVersion, state.CollectionStateOkay, version.Short)
 
+	roles, err := GetRoles(ctx, db)
+	if err != nil {
+		return CollectionHelper{}, fmt.Errorf("failed collecting pg_roles: %s", err)
+	}
+
+	roleByName := make(map[string]state.PostgresRole)
+	roleByOid := make(map[state.Oid]state.PostgresRole)
+	for _, role := range roles {
+		roleByName[role.Name] = role
+		roleByOid[role.Oid] = role
+	}
+
+	collectorRole := roleByName[server.Config.GetDbUsername()]
+	memberOf := collectorRole.MemberOf
+	for _, m := range memberOf { // Allow one level of indirect role memberships
+		memberOf = append(memberOf, roleByOid[m].MemberOf...)
+	}
+	connectedAsSuperUser := collectorRole.SuperUser ||
+		slices.Contains(memberOf, roleByName["rds_superuser"].Oid) ||
+		slices.Contains(memberOf, roleByName["azure_pg_admin"].Oid) ||
+		slices.Contains(memberOf, roleByName["cloudsqlsuperuser"].Oid)
+	connectedAsMonitoringRole := slices.Contains(memberOf, roleByName["pg_monitor"].Oid)
```

Comment:
Per PR review, preference to keep a database connection check here (vs implementing our own pg_has_role).

Reply:
Yeah, I think doing this once is great, but doing it in Go does not seem like a good trade-off to me.